src/services/scenes.ts

```diff
@@ -62,14 +62,10 @@ export function getLabelOptions(sceneObject: SceneObject, allOptions: string[])
  * Given an array of label, or fields names, return those that are not already present in the filters.
  */
 export function getUniqueFilters(sceneObject: SceneObject, labelNames: string[]) {
-  const labelFilters = sceneGraph.lookupVariable(VAR_FILTERS, sceneObject);
-  const fieldsFilters = sceneGraph.lookupVariable(VAR_FIELDS, sceneObject);
+  const labelFilters = sceneGraph.lookupVariable(VAR_FILTERS, sceneObject) as AdHocFiltersVariable;
+  const fieldsFilters = sceneGraph.lookupVariable(VAR_FIELDS, sceneObject) as AdHocFiltersVariable;
```

Comment:
Is this change correct? can't we get undefined here?

Reply:
I'd be confident we should, but we can add an extra check just in case. The important part was to not test for `instanceof` because it would make testing this harder.

Reply:
Done.